staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -77,8 +77,8 @@ func NewFieldManager(f Manager) *FieldManager {
 
 // NewDefaultFieldManager creates a new FieldManager that merges apply requests
 // and update managed fields for other types of requests.
-func NewDefaultFieldManager(models openapiproto.Models, objectConverter runtime.ObjectConvertor, objectDefaulter runtime.ObjectDefaulter, objectCreater runtime.ObjectCreater, kind schema.GroupVersionKind, hub schema.GroupVersion) (*FieldManager, error) {
-	f, err := NewStructuredMergeManager(models, objectConverter, objectDefaulter, kind.GroupVersion(), hub)
+func NewDefaultFieldManager(models openapiproto.Models, objectConverter runtime.ObjectConvertor, objectDefaulter runtime.ObjectDefaulter, objectCreater runtime.ObjectCreater, kind schema.GroupVersionKind, hub schema.GroupVersion, resetFields map[fieldpath.APIVersion]*fieldpath.Set) (*FieldManager, error) {
```

Comment:
I did mention it somewhere on @nodo's PR, but we're crossing the limit where the number of parameter here actually makes sense. Considering that tests have different ways of building that and some fields are sometimes left empty depending on what we're trying to do, this seems like the pattern we initially used (and made sense at that time) is not suited anymore.

Since this is strictly orthogonal, I think it's fine to leave it like this for now (this will still conflict with Andrea's PR but we should absolutely go over that after and find a better pattern (builder pattern?).

Reply:
Couldn't agree more.
Didn't want to bloat this pr with it, but yeah builder might be an option. Not sure yet.
Most params are mandatory, but maybe they could be gut into smaller pieces as well.